### PR TITLE
extend the API to generate diff reports

### PR DESCRIPTION
**Pull request by cosletmihai**

### CODE REVIEW

File: resources/eds_templates/diff_report/config.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+    "template": "main.html",
+    "conf":
+    {
+        "default_endpoint" : "http://fuseki:3030/eurovoc-fragment/sparql",
```
Comment (costezki) [inline resources/eds_templates/diff_report/config.json:5]: this cannot be hardcoded. endpooint is always variable./

File: resources/eds_templates/diff_report/config.json [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+        "title": "Dataset Fingerprint Report (Diff)",
```
Comment (costezki) [inline resources/eds_templates/diff_report/config.json:6]: we deal with a Diff report, this config file needs revision

File: resources/eds_templates/diff_report/templates/macros.html [hunk 3]
```diff
@@ -0,0 +1,57 @@
+{# Default way of showing errors #}
```
Comment (costezki) [inline resources/eds_templates/diff_report/templates/macros.html:1]: we need to investigate how can we provide mactos libraries to the eds4jinja reports. So we no longer carry this macros file around. New jira ticket I guess

File: utils/generate_utils.py [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+# generate_utils.py
+# Date:  24/09/2020
+# Author: Mihai Coșleț
+# Email: coslet.mihai@gmail.com
+
+"""
+Module description
+
+"""
+from pathlib import Path
```
Comment (costezki) [inline utils/generate_utils.py:12]: is that a top level utils folder? If so, we shall do something about it. Moreover this util looks in fact more as a part of the Report generation functionality. Consider refactoring it please.

**Review by costezki — APPROVED** [review]

**Comment by cosletmihai** [comment]
find out how to access templateurile din `eds4jinja2` pentru **rdf diff** report

File: rdf_differ/entrypoints/api/handlers.py [hunk 5]
```diff
@@ -109,3 +117,26 @@ def delete_diff(dataset_id: str) -> tuple:
         return f'<{dataset_id}> created successfully.', 200
     except FusekiException:
         raise NotFound(f'<{dataset_id}> does not exist.')  # 404
+
+
+def get_report(dataset_url: str) -> tuple:
```
[the remark this answers is not in the document]
Comment (cosletmihai) [reply rdf_differ/entrypoints/api/handlers.py:122]: what needs to be done to cover the slow cases?

File: rdf_differ/entrypoints/api/openapi/openapi.yaml [hunk 6]
```diff
@@ -34,7 +34,7 @@ paths:
               schema:
                 type: array
                 items:
-                  $ref: '#/components/schemas/CalculatedDiffDescription'
+                  $ref: "#/components/schemas/CalculatedDiffDescription"
```
[the remark this answers is not in the document]
Comment (cosletmihai) [reply rdf_differ/entrypoints/api/openapi/openapi.yaml:37]: it works for most cases. the case in which it was not working was for a `multipart/form-data` post request

File: resources/eds_templates/diff_report/config.json (hunk 1 above)
Comment (cosletmihai) [reply resources/eds_templates/diff_report/config.json:5]: it is not hardcoded, it's the example I pulled from the `eds4jinja2`

File: utils/generate_utils.py (hunk 4 above)
Comment (cosletmihai) [reply utils/generate_utils.py]: what do you mean by "top level utils" folder?

File: tests/unit/test_generate_report_utils.py [hunk 7]
```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/python3
+
+# test_generate_report_utils.py
+# Date:  24/09/2020
+# Author: Mihai Coșleț
+# Email: coslet.mihai@gmail.com
+from utils.generate_report_utils import generate_config_content
```
Comment (costezki) [inline tests/unit/test_generate_report_utils.py:7]: let's refactor this

**Review by costezki — APPROVED** [review]
Let's move the
```generate_config_content```
and merge the code.

**Review by costezki — COMMENTED** [review]
well done

**Pull request merged**